src/dbnode/server/server.go

```diff
@@ -409,11 +409,26 @@ func Run(runOpts RunOptions) {
 	defer stopReporting()
 
 	// Setup query stats tracking.
+	var statsOpts stats.QueryStatsOptions
+	if runOpts.Config.Limits.MaxRecentlyQueriedBlocks == nil {
+		statsOpts = stats.QueryStatsOptions{
+			Lookback: stats.DefaultLookback,
+		}
+	} else {
+		maxQueriedBlocks := runOpts.Config.Limits.MaxRecentlyQueriedBlocks
+		statsOpts = stats.QueryStatsOptions{
+			MaxDocs:  maxQueriedBlocks.Value,
+			Lookback: maxQueriedBlocks.Lookback,
+		}
+	}
 	var tracker stats.QueryStatsTracker
 	if runOpts.QueryStatsTrackerFn == nil {
-		tracker = stats.DefaultQueryStatsTrackerForMetrics(iopts)
+		tracker, err = stats.DefaultQueryStatsTracker(iopts, statsOpts)
+		if err != nil {
+			logger.Fatal("could not construct query stats tracker from config", zap.Error(err))
+		}
 	} else {
-		tracker = runOpts.QueryStatsTrackerFn(iopts)
+		tracker = runOpts.QueryStatsTrackerFn(iopts, statsOpts)
```

Comment:
Would it be worth adding a `Validate()` method on statsOpts and call it to verify the options look good?